datalad/support/exceptions.py

```diff
@@ -10,9 +10,153 @@
 """
 
 import re
+import sys
+import traceback
 from os import linesep
+from pathlib import Path
 from pprint import pformat
 
+
+class CapturedException(object):
+    """This class represents information about an occurred exception (including
+    its traceback), while not holding any references to the actual exception
+    object or its traceback, frame references, etc.
+
+    Just keep the textual information for logging or whatever other kind of
+    reporting.
+    """
+
+    @classmethod
+    def capture_exc(cls, exc=None, limit=None, capture_locals=False):
+        """Capture an exception and its traceback for logging.
+
+        Clears the exception's traceback frame references afterwards.
+
+        Parameters
+        ----------
+        exc: Exception or None
+          If None, rely on sys.exc_info() to get the latest.
+        limit: int
+          Note, that this is limiting the capturing of the exception's
+          traceback depth. Formatting for output comes with it's own limit.
+        capture_locals: bool
+          Whether or not to capture the local context of traceback frames.
+
+        Returns
+        -------
+        CapturedException
+        """
+        # Note, that with lookup_lines=False the lookup is deferred,
+        # not disabled. Unclear to me ATM, whether that means to keep frame
+        # references around, but prob. not. TODO: Test that.
+
+        if exc is None:
+            exctype, value, tb = sys.exc_info()
+            tb_exc = traceback.TracebackException(
+                exctype, value, tb,
+                limit=limit,
+                lookup_lines=True,
+                capture_locals=capture_locals
+            )
+            traceback.clear_frames(tb)
+        else:
+            tb_exc = traceback.TracebackException.from_exception(
+                exc,
+                limit=limit,
+                lookup_lines=True,
+                capture_locals=capture_locals
+            )
+            traceback.clear_frames(exc.__traceback__)
+
+        return CapturedException(tb_exc)
```

Comment:
if this is the only way to instantiate this class (from an exception), why bother with this `capture_exc` `@classmethod` and not just to make it into `__init__(self, exc, ...)`?

Reply:
>  why not just to make exc_str return the string representation of your CapturedException thus avoiding all those str(exc_str(exc))

Mainly to make sure, that I discover what exactly (Formatters - what else?) would need to deal with an instance of `CapturedException`. Furthermore - thinking of further RF'ing - from my POV those all almost always the wrong places to decide how something should be formatted. So, I do want to break that pattern, when I replace the calls to `exc_str` entirely. Formatting decisions should ultimately be made higher up the chain, while in those places just the object should be passed.

Reply:
> why bother with this capture_exc @classmethod and not just to make it into __init__(self, exc, ...)?

Right. Original idea was somewhat different, but now - will do.

Reply:
Did remove `capture_exc`.

Re `%s` vs `%r`: Did that, too. However, regarding reducing the diff this is not gonna do much. All other places are status dicts where a string is needed (as opposed to something that can be converted to a string).

Reply:
> why to not chase all exc_str invocations

I *do* want to chase them, but with more in mind and therefore in a separate PR - see #5716.